Amazon.IonObjectMapper.Demo/PropertyNameCaseInsensitiveTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * Copyright (c) Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.IonObjectMapper.Demo
+{
+    using System;
+    using System.IO;
+    using Amazon.IonDotnet.Tree;
+    using Amazon.IonObjectMapper.Test;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    public class NormalCaseClass 
+    {
+        public int? A { get; set; }
+        
+        public int? B { get; set; }
+
+        [IonField]
+        public int C;
+    }
+
+    public class AbnormalCaseClass 
+    {
+        public int? a { get; set; }
+        
+        public int? B { get; set; }
+
+        [IonField]
+        public int c;
+    }
+
+    [TestClass]
+    public class PropertyNameCaseInsensitiveTest
+    {
+        [TestMethod]
+        public void Scratch()
+        {
+            NormalCaseClass normalCaseObject = new NormalCaseClass { A = 1, C = 3 };
+
+            AbnormalCaseClass abnormalCaseObject;
+            MemoryStream stream;
+            IonSerializer ionSerializer;
+            
+            ionSerializer = new IonSerializer(new IonSerializationOptions { PropertyNameCaseInsensitive = false });
+            stream = (MemoryStream)ionSerializer.Serialize(normalCaseObject);
+            Console.WriteLine(Utils.PrettyPrint(stream));
+            // {
+            //     a: 1,
```

Comment:
Since we have set `PropertyNameCaseInsensitive` to false, which means the property name is case sensitive now, seems the property name should be `A` instead of `a` as it is defined in `NormalCaseClass`?

Reply:
`PropertyNameCaseInsensitive` affects the strictness of deserialization, not output formatting. The output case is governed by `NamingConvention`, which is `CamelCaseNamingConvention` by default.

The reason the class used for serialization (`NormalCaseClass`) and deserialization (`AbnormalCaseClass`) are different is that, in order to demonstrate the increased flexibility of access offered by setting `PropertyNameCaseInsensitive` to true, the deserialization class needs to have differently-cased properties from the serialization class.

I removed the `WriteLine` statements before deserializing as they detract from the focus on the demo.

Reply:
Instead of using a different class for deserialization, we could demo the use case of `PropertyNameCaseInsensitive` option by deserializing from an Ion text that has different casing than the serialized class. 

For example, if the C# property name is `TitleCasing` and the Ion text is `{TiTLeCASINg:"demo"}`, the ObjectMapper could map the Ion field `TiTLeCASINg` to C# property `TitleCasing` during deserialization when `PropertyNameCaseInsensitive` is set to `true`.

Reply:
Changed the test to load an `ionText` string and show the difference in deserializing with/without `PropertyNameCaseInsensitive`